### PR TITLE
feat: add linters

### PR DESCRIPTION
**Pull request by MSevey**
## Overview

Closes: #372 

This PR: 

- adds some new Makefile commands for runner linters and formatting
- enables lint in CI
- fixes some of the lint errors

NOTE: the markdownlint link checker is failing on linking to /e2e because there is nothing in the directory except the subdirectory, it actually 404s.  This PR will actually resolve that if it gets merged: https://github.com/celestiaorg/knuu/pull/404

## Summary by CodeRabbit

- **New Features**
  - Introduced a GitHub Actions workflow for linting, including checks for Go code, YAML files, Markdown files, and Go package vulnerabilities.

- **Documentation**
  - Improved the contribution guidelines for better clarity and readability.
  - Updated the README to include a hyperlink for the documentation.

- **Chores**
  - Added exclusions to the `.gitignore` file for specific files and directories.
  - Reorganized import statements across multiple files for better code organization.
  - Updated the Makefile with new targets for various tasks like cleaning, testing, and linting.

- **Bug Fixes**
  - Improved error handling in multiple test functions by changing `t.Fatalf` to `t.Errorf`.
  - Corrected the handling of time tickers in the `WaitInstanceIsRunning` method.

### CODE REVIEW

File: pkg/builder/kaniko/utils_test.go [hunk 1]
```diff
@@ -22,10 +22,10 @@ func TestCreateTarGz(t *testing.T) {
 	}
 	for file := range expectedFiles {
 		filePath := filepath.Join(testDir, file)
-		err := os.MkdirAll(filepath.Dir(filePath), os.ModePerm)
+		err := os.MkdirAll(filepath.Dir(filePath), 0600)
```
Comment (mojtaba-esk) [inline pkg/builder/kaniko/utils_test.go:25]: why instead of predefined constant a hard coded value is used?

**Review by mojtaba-esk — COMMENTED** [review]
lgtm :rocket:  
left a question.

File: pkg/builder/kaniko/utils_test.go (hunk 1 above)
Comment (MSevey) [reply pkg/builder/kaniko/utils_test.go]: Defined one locally since there are not constants defined in the go libraries.

Comment (mojtaba-esk) [reply pkg/builder/kaniko/utils_test.go]: Thanks.
I see. that `os.ModePerm` is I think `0777` like full perms.

**Review by mojtaba-esk — APPROVED** [review]
LGTM!

File: pkg/builder/kaniko/utils_test.go (hunk 1 above)
Comment (MSevey) [reply pkg/builder/kaniko/utils_test.go]: correct, but that is the only one. They don't have all the combos.

**Pull request merged**